hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -210,23 +209,24 @@ private OMResponse processReply(OMRequest omRequest, RaftClientReply reply)
   }
 
   /**
-   * Parse errorMessage received from the exception and convert to
-   * {@link OzoneManagerProtocolProtos.Status}.
-   * @param errorMessage
-   * @return OzoneManagerProtocolProtos.Status
+   * Convert exception to {@link OzoneManagerProtocolProtos.Status}.
+   * @param cause - Cause from stateMachine exception
+   * @return {@link OzoneManagerProtocolProtos.Status}
    */
-  private OzoneManagerProtocolProtos.Status parseErrorStatus(
-      String errorMessage) {
-    if (errorMessage.contains(STATUS_CODE)) {
-      String errorCode = errorMessage.substring(
-          errorMessage.indexOf(STATUS_CODE) + STATUS_CODE.length());
-      LOG.debug("Parsing error message for error code " +
-          errorCode);
-      return OzoneManagerProtocolProtos.Status.valueOf(errorCode.trim());
+  private OzoneManagerProtocolProtos.Status exceptionToResponseStatus(
+      Throwable cause) {
+    if (cause == null) {
+      // Current Ratis is setting cause, this is an safer side check.
```

Comment:
I think we would never hit this code. In line 188 (before calling this function) we call "stateMachineException.getCause().getMessage()". If the cause was null, we would get a NullPointerException there itself.

Reply:
Fixed.